### ISSUE TITLE
Consider deprecating Base64Url

### ISSUE THREAD

**Issue by joegoldman2** [passwordless-lib/fido2-net-lib#546]
.NET 9 adds support for Base64 Url (https://github.com/dotnet/runtime/issues/1658). It's also compatible with .NET Standard 2.0. The class Base64Url should probably be deprecated in favor of the built-in support in the BCL.

**Comment by iamcarbon** [comment]
Agreed. We should consider taking a dependency on Microsoft.Bcl.Memory for .NET8.0, and dropping our custom Base64Url reader once it is released in November. The new implementation is vectorized and fuzzed.

We may also want to consider taking a dependency on System.Text.Json 9.0 (which cross-targets .NET8.0) after RTM to utilize the built in JsonStringEnumConverter.

**Comment by abergs** [comment]
Seems like a good idea.

**Comment by abergs** [comment]
While .NET 9 has now been shipped, I don't think we want to remove Base64Url and force every consumer to upgrade to .net9 as .net 8 is still lts.

Perhaps this would be a good v5 issue?

**Comment by abergs** [comment]
So you mean if we upgrade Microsoft.Bcl.Memory we can remove our Base64Url implementation?

**Comment by joegoldman2** [comment]
Yes, we just need to upgrade it and use the provided Base64Url implementation. There is no need to update the target framework of Fido2 lib.

**Comment by iamcarbon** [comment]
Implemented in #575
